File: src/tabakon.net/AngularApp/src/app/models/RetailEndpoint.ts

export class RetailEndpointExtData{
    public retailVersion? : RetailVersion
    public retailExtConfiguration? : RetailExtConfiguration
}

export class RetailEndpoint{
    public retailEndpointIdentity : string;
    public retailEndpointName : string;
    public retailEndpointHost : string;
    public retailEndpointUrl : string;
    public markAsDeleted : string;

    public extData : RetailEndpointExtData;
}


class AbstractCacheEntity{
    public lastCheck : Date
    public retailEndpointIdentity : string;
    public jsonData : string;
}

export class RetailVersion extends AbstractCacheEntity{

}

export class RetailExtConfiguration extends AbstractCacheEntity{

}

File: src/tabakon.net/Tabakon.Entity/RetailEndpoint.cs
using System;

namespace Tabakon.Entity
{
    public class RetailEndpoint
    {
        public string RetailEndpointIdentity {get;set;}
        public string RetailEndpointName {get;set;}
        public string RetailEndpointHost {get;set;}
        public string RetailEndpointUrl {get;set;}
        public bool MarkAsDeleted { get; set; }
        public string StoreRef { get; set; }
    }
}
